internal/datastore/crdb/pool/pool.go

```diff
@@ -141,12 +142,16 @@ func (p *RetryPool) ID() string {
 
 // MaxConns returns the MaxConns configured on the underlying pool
 func (p *RetryPool) MaxConns() uint32 {
-	return uint32(p.pool.Config().MaxConns)
+	// This should be non-negative
+	maxConns, _ := safecast.ToUint32(p.pool.Config().MaxConns)
```

Comment:
If we have enough of these examples, we should add our own MustToUint32 method and just have it panic if it fails